test/e2e/network/ingress.go

```diff
@@ -733,6 +733,148 @@ var _ = SIGDescribe("Loadbalancing: L7", func() {
 			By("Scale down number of backends to 2")
 			scaleAndValidateExposedNEG(3)
 		})
+
+		It("should not delete NEGs when switching between Ingress-referenced NEG and standalone NEG", func() {
+			By("Create a basic HTTP ingress using standalone NEG")
+			jig.CreateIngress(filepath.Join(ingress.IngressManifestPath, "neg-exposed"), ns, map[string]string{}, map[string]string{})
+			jig.WaitForIngress(true)
+			usingNEG, err := gceController.BackendServiceUsingNEG(jig.GetServicePorts(false))
+			Expect(err).NotTo(HaveOccurred())
+			Expect(usingNEG).To(BeTrue())
+
+			name := "hostname"
+			svc, err := f.ClientSet.CoreV1().Services(ns).Get(name, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			var status ingress.NegStatus
+			v, ok := svc.Annotations[ingress.NEGStatusAnnotation]
+			Expect(ok).To(BeTrue())
+
+			err = json.Unmarshal([]byte(v), &status)
+			Expect(err).NotTo(HaveOccurred())
+			framework.Logf("Got %v: %v", ingress.NEGStatusAnnotation, v)
+
+			// Expect 2 NEGs to be created based on the test setup (neg-exposed)
+			if len(status.NetworkEndpointGroups) != 2 {
+				framework.Logf("Expected 2 NEGs, got %d", len(status.NetworkEndpointGroups))
+				Expect(len(status.NetworkEndpointGroups)).To(Equal(2))
+			}
+
+			// Disable NEG exposed but enable NEG with Ingress. Check NEGs still exist
+			By("Switch backend service to use Ingress-exposed NEG")
+			svcList, err := f.ClientSet.CoreV1().Services(ns).List(metav1.ListOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			for _, svc := range svcList.Items {
+				svc.Annotations[ingress.NEGAnnotation] = `{"ingress": true}`
+				_, err = f.ClientSet.CoreV1().Services(ns).Update(&svc)
+				Expect(err).NotTo(HaveOccurred())
+			}
+			wait.Poll(5*time.Second, framework.LoadBalancerPollTimeout, func() (bool, error) {
+				svc, err := f.ClientSet.CoreV1().Services(ns).Get(name, metav1.GetOptions{})
+				if err != nil {
+					return false, nil
+				}
+
+				var status ingress.NegStatus
+				v, ok := svc.Annotations[ingress.NEGStatusAnnotation]
+				if !ok {
+					// Wait for NEG sync loop to find NEGs
+					framework.Logf("Waiting for %v, got: %+v", ingress.NEGStatusAnnotation, svc.Annotations)
+					return false, nil
+				}
+
+				err = json.Unmarshal([]byte(v), &status)
+				if err != nil {
+					framework.Logf("Error in parsing Expose NEG annotation: %v", err)
+					return false, nil
+				}
+				framework.Logf("Got %v: %v", ingress.NEGStatusAnnotation, v)
+
+				// Expect 2 NEGs to be created based on the test setup (neg-exposed)
+				if len(status.NetworkEndpointGroups) != 2 {
+					framework.Logf("Expected 2 NEGs, got %d", len(status.NetworkEndpointGroups))
+					return false, nil
+				}
+
+				return gceController.BackendServiceUsingNEG(jig.GetServicePorts(false))
+			})
+			jig.WaitForIngress(true)
+		})
+		It("should delete NEGs when removing standalone NEG notation", func() {
```

Comment:
I feel like a seperate test just for ensuring the NEGs are deleted is a bit heavy. 

Can we add an extra step on the above test?

1. Initially
```
'{"ingress":true,"exposed_ports":{"80":{},"443":{}}}'
```
Expect 2 NEGs AND Ingress using NEG
2. 
```
'{"ingress":true,"exposed_ports":{"443":{}}}'
```
Expect 2 NEGs AND Ingress using NEG. 
Observe for a while and check if there is any NEG deletion events associated with the test service. (Maybe there is a better way to ensure NEG controller did not attempt to delete it?)
3. 
```
'{"ingress":false,"exposed_ports":{"443":{}}}'
```
Ensure NEG deleted and Ingress no longer using NEG backend

4. 
```
'{"ingress":false"}'
```
Expect 0 NEG left.